packages/react-cookie/src/Cookie.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import {useRequestHeader} from '@shopify/react-network';
+import {useLazyRef} from '@shopify/react-hooks';
+import {useSerialized} from '@shopify/react-html';
+import {CookieManager} from './manager';
+import {CookieContext} from './context';
+import {hasDocumentCookie} from './utilities';
+import {Cookies} from './types';
+
+export interface Props {
+  children?: React.ReactNode;
+}
+
+export function Cookie({children}: Props) {
+  const [initialCookies, Serialize] = useSerialized<Cookies>('cookies');
+  const serverCookieHeader = useRequestHeader('cookie');
+
+  const manager = useLazyRef(() => {
+    const cookies = hasDocumentCookie() ? document.cookie : serverCookieHeader;
+
+    return new CookieManager(initialCookies || cookies);
+  }).current;
+
+  return (
+    <>
+      <CookieContext.Provider value={manager}>
+        {children}
+      </CookieContext.Provider>
+      <Serialize data={() => manager.getCookies()} />
```

Comment:
I don't understand this 😕 Why wouldn't cookies be sent as response headers?

Reply:
The `useRequestHeader ` hook returns `undefined` on the client so without this, the server cookies are lost.

Reply:
I haven't worked much with cookies, but as I understand it, this should be possible:
- The server side of this generates `Set-Cookie` headers for each name/value pair
- The browser sees these headers, and populates `document.cookie` to reflect them
- The client side of this grabs `document.cookie`, and parses it into name/value pairs

Not an expert on this, so I could definitely be wrong.

Reply:
The above seems right